services/web-app/layouts/layout/layout.js

```diff
@@ -56,12 +88,27 @@ const Layout = ({ children }) => {
   }, [])
 
   useEffect(() => {
-    setShowSideNav(!router.pathname.startsWith('/assets/[host]/[org]/[repo]/[library]/[ref]'))
-  }, [router.pathname])
+    setShowSideNav(
+      !router.pathname.startsWith('/assets/[host]/[org]/[repo]/[library]/[ref]/[asset]')
+    )
+    setLibrarySideNav(router.pathname.startsWith('/assets/[host]/[org]/[repo]/[library]/[ref]'))
+    setLibraryNavSlideOut(false)
+  }, [setShowSideNav, setLibrarySideNav, setLibraryNavSlideOut, router.pathname])
+
+  const onClickSideNavExpand = () => {
+    toggleSideNavExpanded(!isSideNavExpanded)
+  }
+
+  const backLink = () => {
+    setLibraryNavSlideOut(true)
+    setTimeout(() => router.push('/assets/libraries'), 150)
```

Comment:
minor: magic number could be defined as a constant at the top describing its purpose.